libvast/src/system/archive.cpp

```diff
@@ -83,6 +48,67 @@ void archive_state::send_report() {
   self->send(accountant, std::move(r));
 }
 
+std::unique_ptr<segment_store::lookup> next_session(archive_state& state) {
+  while (!state.requests.empty()) {
+    auto& request = state.requests.front();
+    if (request.cancelled || request.ids_queue.empty()) {
+      // Not sure whether this is really necessary.
+      while (!request.ids_queue.empty()) {
+        request.ids_queue.front().second.deliver(atom::done_v);
+        request.ids_queue.pop();
+      }
+      state.requests.pop_front();
+      continue;
+    }
+    // We found an active request.
+    auto& [ids, rp] = request.ids_queue.front();
+    state.active_promise = std::move(rp);
+    state.session_ids = std::move(ids);
+    request.ids_queue.pop();
+    return state.store->extract(state.session_ids);
+  }
+  VAST_TRACE("archive has no requests");
+  return nullptr;
+}
+
+auto file_request(archive_actor::stateful_pointer<archive_state> self,
```

Comment:
Please make this a member function of the archive state and document it accordingly.